k2/csrc/ragged_ops_inl.h

```diff
@@ -123,16 +150,14 @@ Ragged<T> Transpose(Ragged<T> &src) {
   int32_t **ans_row_ids_data = ans_row_ids_ptr.Data();
   Array1<int32_t *> src_row_splits_ptr = GetRowSplitsPtr(src_shape);
   int32_t **src_row_splits_data = src_row_splits_ptr.Data();
-  int32_t num_axes = src.NumAxes();
   Array1<T> ans_values(c, src_values.Dim());
   const T *src_data = src_values.Data();
   T *ans_data = ans_values.Data();
   auto lambda_set_values = [=] __host__ __device__(int32_t new_idx) {
-    int32_t *idx_data =
-        reinterpret_cast<int32_t *>(malloc(sizeof(int32_t) * num_axes));
+    int32_t idx_data[kMaxNumAxes];
```

Comment:
Replace dynamic malloc with a static array. For now I suppose max-num-axes <= 6 (actually the code just requires size 
 with NumAxes() - 1)

Reply:
That must have slipped past review last time, I would have objected if I saw the malloc in there.
Even having every warp doing malloc would be super slow.  Every thread is a complete no-go .

Reply:
Yeah, that's why I update it this time. So do you think if the current approach is fine or not? Or any other suggestion?